Library/Homebrew/dev-cmd/test-bot.rb

```diff
@@ -920,7 +920,6 @@ def sanitize_ARGV_and_ENV
     end
 
     ENV["HOMEBREW_DEVELOPER"] = "1"
-    ENV["HOMEBREW_NO_AUTO_UPDATE"] = "1"
```

Comment:
I'm confused. How does removing this from the `brew test-bot` code ensure the auto-update doesn't later mess with the state of the repositories once `brew test-bot` has pulled some PR commits or similar activities that involve temporarily modifying the local repository and creating a divergent `master` branch?

From a typical run of `brew test-bot`:

```
# This modifies the master branch locally, making it diverge from upstream:
==> brew pull --clean https://github.com/Homebrew/homebrew-core/pull/3704
[…]
# This causes the auto-update to kick in if it hasn't been disabled in test-bot:
==> brew install --only-dependencies --build-bottle --verbose mesos
```